### PR TITLE
Owasp top10 2017 apps/a8/amarelo designs

### PR DESCRIPTION
**Pull request by thiagolotufo**
## This solution refers to which of the apps?

A/M# - owasp-top10-2017-apps/a8/amarelo-designs

## What did you do to mitigate the vulnerability?

The pickle should only be used when we have a reliable source accessing that information. In this lab, we are using pickle to serialize a cookie, and this is not safe.
To mitigate this vulnerability, I replaced the pickle with JWT, to generate the session token.

## Did you test your changes? What commands did you run?

To mitigate this vulnerability, I used JWT to create and validate the user's session token.
To test, I used the deserialization file created for the pickle. When we type the code on the command line, it does not return any information, as it does not have access to the application.

- Vulnerability test

Screen Shot 2021-04-15 at 17 23 21

The token that the command returns is generated in the code itself, and is not retrieved from the http request.

### CODE REVIEW

File: owasp-top10-2017-apps/a8/amarelo-designs/app/app.py [hunk 1]
```diff
@@ -18,12 +21,14 @@ def login():
         password = request.values.get('password')
     
         if username == "admin" and password == "admin":
```
Comment (gustavocovas) [inline owasp-top10-2017-apps/a8/amarelo-designs/app/app.py:23]: Be careful with hardcoded passwords!

**Review by gustavocovas — CHANGES_REQUESTED** [review]
Nice job, @thiagolotufo! After your changes, I couldn't reproduce the attack narrative anymore

But there is still a hardcoded password in the source code. Can you configure it with an environment variable instead?

**Comment by thiagolotufo** [comment]
@gustavocovas

The application is now using the environment variable for the password.
Can you verify that the vulnerabilities have been mitigated?

**Review by gustavocovas — CHANGES_REQUESTED** [review]
Nice, but I think we should use an environment variable to configure the JWT secret as well

**Comment by thiagolotufo** [comment]
> Nice, but I think we should use an environment variable to configure the JWT secret as well (we don't want to have any secrets in the source code)

You're right @gustavocovas. 
I made some changes so that the secret of JWT is an environment variable.

**Pull request closed without merging**